src/app/components/FetchCryptoData.tsx

```diff
@@ -1,31 +1,53 @@
 import React from 'react'
 
+import CryptoCard from './CryptoCard';
+import UseCoinGeckoApi from './UseCoinGeckoApi'; 
+
+
 const FetchCryptoData = () => {
-    const [state, setState] = React.useState({
-        data: null,
-        error: null
-    });
-
-    React.useEffect(() => {
-        fetch("https://api.coingecko.com/api/v3/coins/markets?vs_currency=usd")
-          .then((response) => response.json())
-          .then(
-            (data) => setState({ ...state, data: data }),
-            (error) => setState({ ...state, error: error })
-          );
-    }, []);
-
-    if (state.error) {
-        return <div>Error: {state.error.message}</div>;
-    } else {
-        return (
+    
+    const { data: state, error: error } = UseCoinGeckoApi();
+
+    const [search, setSearch] = React.useState("");
+
+    const handleChange = (e) => {
+        setSearch(e.target.value);
+    }
+
+    const filteredCoins = state && state.filter((coin) =>
+        coin.name.toLowerCase().includes(search.toLowerCase())
+    );
+
+    if (error) {
+        return <div>Error: {error} </div>;
+    }
+    return (
+        <>
             <div>
-                {state.data && state.data.map((item: any) => (
-                    <div key={item.id}>{item.name}</div>
-                ))}
+                <input type="text" onChange={handleChange} />
             </div>
-        );
-    }
+            <div>
+                {filteredCoins && filteredCoins.map((coin) => {
```

Comment:
I dont think you need this check `filteredCoins &&`. if `filteredCoins` is empty then nothing will render here